v2/contribute/decisions/session/0027-use-standard-jwt-verification-lib-on-BE.mdx

```diff
@@ -0,0 +1,29 @@
+---
+id: "0027"
+title: Use a standard JWT verification lib on the BE
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Use a standard JWT verification lib on the BE
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-06" created="2022-12-06" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Since our access tokens will match standard JWTs, we can switch to using a jwt verification library with key caching instead of our own implementation.
+
+## Considered Options
+
+* Keep using our own implementation
+* Switch to an external library
+
+## Decision Outcome
+
+We should switch to an external library
+- Our code gets smaller
+- We use the same method we will recommend to people if they aren't using our SDKs
```

Comment:
We could merge the JWT recipe into Session, but I don't know if that's necessary. It's a nice division of interfaces this way.
The openID recipe should be kept separate to enable the OpenID config endpoint.
If we assume that the Session recipe is always enabled (or we require that when initializing the other 2) then we don't even have issues with overlapping configuration.

Reply:
Right. But what I was talking about is that if `jwt: enable: true` is done now, it implies that the access token is exposed on the frontend as well. So if we remove this `jwt` config, then the user will have to use the `exposeAccessTokenToFrontendInCookieBasedAuth` config and change how they read the JWT? Anything else required to change?

Reply:
> So if we remove this jwt config, then the user will have to use the exposeAccessTokenToFrontendInCookieBasedAuth config and change how they read the JWT?

Yep.

>Anything else required to change?

There shouldn't be anything because of this decision. Other things like the changes to the JWKs endpoint, and config changes could require some updates.

Reply:
We decided that we will deprecate the jwt: enable: true config and that if that is used, it will be the same as the user setting exposeAccessTokenToFrontendInCookieBasedAuth to true.